test/unittest/dynamic/discovery_database/DiscoveryDatabaseTest.cpp

```diff
@@ -0,0 +1,208 @@
+    Guid inactive_guid = random_guid(2);
+    Endpoint inactive_endpoint(EndpointKind::READER, inactive_guid, qos, topic);
+    inactive_endpoint.active(false);
+
+    // Insert endpoints
+    ASSERT_TRUE(discovery_database.add_endpoint(active_endpoint));
+    ASSERT_EQ(discovery_database.get_endpoint(active_guid), active_endpoint);
+    ASSERT_TRUE(discovery_database.add_endpoint(inactive_endpoint));
+    ASSERT_EQ(discovery_database.get_endpoint(inactive_guid), inactive_endpoint);
+
+    // Add new endpoint with same guid as already stored active endpoint
+    // Should throw \c InconsistencyException
+    RealTopic new_topic("new", "new");
+    Endpoint active_new_endpoint(EndpointKind::READER, active_guid, qos, new_topic);
+    ASSERT_THROW(discovery_database.add_endpoint(active_new_endpoint), InconsistencyException);
+
+    // Add new endpoint with same guid as already stored inactive endpoint
+    // Should update entry
+    Endpoint inactive_new_endpoint(EndpointKind::READER, inactive_guid, qos, new_topic);
+    ASSERT_TRUE(discovery_database.add_endpoint(inactive_new_endpoint));
+    ASSERT_EQ(discovery_database.get_endpoint(inactive_guid), inactive_new_endpoint);
+}
+
+/**
+ * Test \c DiscoveryDatabase \c update_endpoint method
+ *
+ * CASES:
+ *  Endpoint to be updated not present in database
+ *  Endpoint to be updated present in database
+ */
+TEST(DiscoveryDatabaseTest, update_endpoint)
+{
+    DiscoveryDatabase discovery_database;
+    Guid guid = random_guid(1);
+    QoS qos;
+    RealTopic topic("original", "original");
+    Endpoint endpoint(EndpointKind::READER, guid, qos, topic);
+    RealTopic new_topic("new", "new");
+    Endpoint new_endpoint(EndpointKind::READER, guid, qos, new_topic);
+    // Endpoint to be updated not yet inserted
+    ASSERT_THROW(discovery_database.update_endpoint(new_endpoint), InconsistencyException);
+
+    // Insert endpoint
+    ASSERT_TRUE(discovery_database.add_endpoint(endpoint));
+    ASSERT_TRUE(discovery_database.topic_exists(topic));
+    ASSERT_FALSE(discovery_database.topic_exists(new_topic));
+    ASSERT_EQ(discovery_database.get_endpoint(guid), endpoint);
+
+    // Update endpoint
+    ASSERT_TRUE(discovery_database.update_endpoint(new_endpoint));
+    ASSERT_FALSE(discovery_database.topic_exists(topic));
+    ASSERT_TRUE(discovery_database.topic_exists(new_topic));
+    ASSERT_EQ(discovery_database.get_endpoint(guid), new_endpoint);
+}
+
+/**
+ * Test \c DiscoveryDatabase \c erase_endpoint method
+ *
+ * CASES:
+ *  Endpoint to erase not present in database
+ *  Endpoint to erase present in database
+ */
+TEST(DiscoveryDatabaseTest, erase_endpoint)
+{
+    DiscoveryDatabase discovery_database;
+    Guid guid = random_guid(1);
+    QoS qos;
+    RealTopic topic("test", "test");
+    Endpoint endpoint(EndpointKind::READER, guid, qos, topic);
+
+    // Endpoint to erase not yet inserted
+    ASSERT_EQ(discovery_database.erase_endpoint(guid), ReturnCode::RETCODE_NO_DATA);
```

Comment:
Discussion: Should this method also throw inconsistency? I think so.

Reply:
:+1: